osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs

```diff
@@ -12,14 +14,31 @@
 using osu.Game.Rulesets.Osu.Objects;
 using osu.Game.Rulesets.Scoring;
 using osu.Game.Scoring;
+using osu.Game.Rulesets.Osu.Difficulty.MathUtil;
 
 namespace osu.Game.Rulesets.Osu.Difficulty
 {
     public class OsuPerformanceCalculator : PerformanceCalculator
     {
         public new OsuDifficultyAttributes Attributes => (OsuDifficultyAttributes)base.Attributes;
 
+        /// <summary>
+        /// Aim, tap and acc values are combined using power mean with this as the exponent.
+        /// </summary>
+        private const double total_value_exponent = 1.5;
+
+        /// <summary>
+        /// This exponent is used to convert throughput to aim pp and tap skill to tap pp.
+        /// </summary>
+        private const double skill_to_pp_exponent = 2.7;
+
+        /// <summary>
+        /// The first 0.5 miss doesn't count when we penalize misses
```

Comment:
*Half* miss? What is a half miss?

Does this mean that the first miss carries half the normal penalty?

Reply:
he attempts to get a lower bound on misses+slider breaks, but it's an approximation because he doesn't want to include missed slider ends, so the approximation varies continuously